src/Update/Nix/FetchGit/Utils.hs

```diff
@@ -83,3 +102,21 @@ parseISO8601DateToDay t =
   case parseTimeM False defaultTimeLocale "%Y-%m-%d" justDate of
     Nothing -> Left $ InvalidDateString t
     Just day -> pure day
+
+instance Show Warning where
```

Comment:
I was looking in to this for a while.  That idiom is unavailable to us because our `Warning` type contains a `Text.PrettyPrint.ANSI.Leijen.Doc` object.  So when GHC derives `Show` for our `Warning`, it ends up running `show` on that doc object, which just returns a pretty-printed version suitable for console output.  So the output looks something like this (which I just made up on the spot):

    CoultNotParseInput foo.nix:14:5: got ',', expected '<' or '>'
      or '}' or '{'

The places that [talk about that idiom](https://hackage.haskell.org/package/base-4.9.0.0/docs/Text-Show.html) only refer to *derived* instances of show, not manually-defined instances.

Still, I don't really care, so I'm happy to make a `prettyShow` function.  I think I would want to remove the `Show` typeclass from `Warning` so that we don't accidentally use it somewhere, and because it doesn't follow that idiom you wanted.